File: mobile/android/app/src/main/java/com/cloison/app/LlmClient.kt
// CLOISON Mobile — client HTTP minimal (POST /chat/completions).
// Le corps reçu est DÉJÀ tokenisé par le WASM côté JS : le fournisseur ne
// voit jamais la PII en clair. Réponse brute renvoyée au JS pour restauration.
package com.cloison.app

import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

object LlmClient {

    fun post(cfg: LlmConfig, bodyJson: String): String {
        val base = cfg.baseUrl.trimEnd('/')
        val url = URL("$base/chat/completions")
        val conn = url.openConnection() as HttpURLConnection
        try {
            conn.requestMethod = "POST"
            conn.connectTimeout = 15_000
            conn.readTimeout = 60_000
            conn.setRequestProperty("Content-Type", "application/json")
            conn.setRequestProperty("Authorization", "Bearer ${cfg.apiKey}")
            conn.doOutput = true
            conn.outputStream.use { it.write(bodyJson.toByteArray(Charsets.UTF_8)) }

            val stream = if (conn.responseCode in 200..299) conn.inputStream else conn.errorStream
            val text = stream?.let {
                BufferedReader(InputStreamReader(it, Charsets.UTF_8)).use { r -> r.readText() }
            }.orEmpty()
            if (conn.responseCode !in 200..299) {
                // Renvoie une erreur JSON lisible par le JS.
                return "{\"error\":{\"message\":\"HTTP ${conn.responseCode}: $text\"}}"
            }
            return text
        } finally {
            conn.disconnect()
        }
    }
}

File: mobile/ios/Cloison/LlmClient.swift
// CLOISON Mobile — client HTTP minimal (POST {base}/chat/completions).
// Le corps reçu est DÉJÀ tokenisé par le WASM côté JS : le fournisseur ne
// voit jamais la PII en clair. Réponse brute renvoyée au JS pour restauration.
//
// Swift Concurrency : tâche async, résultat via Task { @MainActor }.

import Foundation

enum LlmError: LocalizedError {
    case invalidUrl
    case http(Int, String)

    var errorDescription: String? {
        switch self {
        case .invalidUrl:
            return "Base URL invalide"
        case .http(let code, let text):
            return "HTTP \(code): \(text)"
        }
    }
}

enum LlmClient {

    /// POST {base}/chat/completions avec le corps JSON (déjà tokenisé).
    static func post(cfg: LlmConfig, bodyJson: String) async throws -> String {
        guard let base = URL(string: cfg.baseUrl.trimmingCharacters(in: .whitespacesAndNewlines)),
              base.scheme != nil, base.host != nil else {
            throw LlmError.invalidUrl
        }
        let url = base.appendingPathComponent("chat/completions")
        var req = URLRequest(url: url)
        req.httpMethod = "POST"
        req.timeoutInterval = 60
        req.setValue("application/json", forHTTPHeaderField: "Content-Type")
        req.setValue("Bearer \(cfg.apiKey)", forHTTPHeaderField: "Authorization")
        req.httpBody = bodyJson.data(using: .utf8)

        let (data, resp) = try await URLSession.shared.data(for: req)
        guard let http = resp as? HTTPURLResponse else {
            throw LlmError.http(0, "réponse non-HTTP")
        }
        let text = String(data: data, encoding: .utf8) ?? ""
        guard (200...299).contains(http.statusCode) else {
            throw LlmError.http(http.statusCode, text)
        }
        return text
    }
}
